spec/factories/users.rb

```diff
@@ -8,11 +8,11 @@
 
   factory :user do
     name                         { Faker::Name.name }
-    email                        { generate :email }
-    username                     { generate :username }
+    email                        { Faker::Internet.email }
+    username                     { Faker::Internet.username }
```

Comment:
this won't work, as username has a few rules:

1. can't be one of `ReservedWords.all` (which also includes all the routes)
1. can only be `/\A[a-zA-Z0-9_]+\z/` but `Faker::Internet.username` generates sepators like dash or dot, maybe with `Faker::Internet.username(separators: [])` you can go on a bit, see https://github.com/faker-ruby/faker/blob/master/doc/default/internet.md

same thing for `.email`

There's still the fact that these names will be random, so Percy would fail anyway? Maybe I didn't understand fully the scope of the problem.

One last thing: as we generate thousands of user objects, there' still the chance that a collision can occurr, not likely but also not impossible. I guess that's the reason they decided to use [factory bot sequences](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#sequences).

See the implementation of `Faker::Internet.username`:

https://github.com/faker-ruby/faker/blob/a7aa80877f7bcad435b9a19a62cf5a82b5f449b0/lib/faker/default/internet.rb#L35-L74

Reply:
@rhymes yeah, i clearly got too excited late last night when I pushed this up and was hopeful that I could quickly fix this (LOL 😭)

FWIW, we _should_ be able to assert that the same value will be deterministically generated by Faker (at least, in the same _order_) because of [this](https://github.com/thepracticaldev/dev.to/blob/master/spec/rails_helper.rb#L158-L168) bit in our `rails_helper.rb`.